authentication/account/worker/user_deactivation_notification_worker_test.go

```diff
@@ -0,0 +1,127 @@
+	config.GetUserDeactivationFetchLimitFunc = func() int {
+		return 100
+	}
+	config.GetUserDeactivationInactivityPeriodDaysFunc = func() time.Duration {
+		return 30 * 24 * time.Hour // 31 days, ie, 7 days after notification
+	}
+	config.GetUserDeactivationInactivityNotificationPeriodDaysFunc = func() time.Duration {
+		return 20 * 24 * time.Hour // 24 days
+	}
+	config.GetPostDeactivationNotificationDelayMillisFunc = func() time.Duration {
+		return 5 * time.Millisecond
+	}
+	ctx := context.Background()
+	// yesterday := time.Now().Add(-1 * 24 * time.Hour)
+	ago40days := time.Now().Add(-40 * 24 * time.Hour) // 40 days since last activity and notified...
+
+	var notificationServiceMock *appservicemock.NotificationServiceMock
+	var app application.Application
+	s.SetupSubtest = func() {
+		notificationServiceMock = appservicemock.NewNotificationServiceMock(s.T())
+		notificationServiceMock.SendMessageAsyncFunc = func(ctx context.Context, msg notification.Message, options ...rest.HTTPClientOption) (r chan error, r1 error) {
+			return nil, nil
+		}
+		app = gormapplication.NewGormDB(s.DB, s.Configuration, s.Wrappers, factory.WithNotificationService(notificationServiceMock))
+	}
+
+	s.Run("one user to notify once", func() {
+		// given
+		user1 := s.Graph.CreateUser().User()
+		identity1 := user1.Identities[0]
+		identity1.LastActive = &ago40days
+		err := s.Application.Identities().Save(ctx, &identity1)
+		require.NoError(s.T(), err)
+		// start the worker with a 50ms ticker
+		w := worker.NewUserDeactivationNotificationWorker(context.Background(), app)
+		w.Start(time.Millisecond * 50)
+		defer w.Stop()
+		// wait a few cycles before checking the results
+		time.Sleep(time.Millisecond * 100)
+		// then load the user and check her deactivation notification status
+		result, err := s.Application.Identities().Load(context.Background(), identity1.ID)
+		require.NoError(s.T(), err)
+		assert.NotNil(s.T(), result.DeactivationNotification)
+		// notification only sent once to the user
+		assert.Equal(s.T(), uint64(1), notificationServiceMock.SendMessageAsyncCounter)
+	})
+
+	s.Run("multiple workers but only one working", func() {
+		// given
+		user1 := s.Graph.CreateUser().User()
+		identity1 := user1.Identities[0]
+		identity1.LastActive = &ago40days
+		err := s.Application.Identities().Save(ctx, &identity1)
+		require.NoError(s.T(), err)
+		// start the workers with a 50ms ticker
+		w1 := s.newUserDeactivationNotificationWorker("pod-1", app)
+		w2 := s.newUserDeactivationNotificationWorker("pod-2", app)
+		w3 := s.newUserDeactivationNotificationWorker("pod-3", app)
+		w4 := s.newUserDeactivationNotificationWorker("pod-4", app)
+		freq := time.Millisecond * 50
+		w1.Start(freq)
+		w2.Start(freq)
+		w3.Start(freq)
+		w4.Start(freq)
+		// wait a few cycles before checking the results
+		time.Sleep(freq * 5)
+		w1.Stop()
+		w2.Stop()
+		w3.Stop()
+		w4.Stop()
+		time.Sleep(freq * 10) // give workers some time to stop for good
```

Comment:
for this type of test, I would use approach using a latch - something like this:
```go
                var starter sync.WaitGroup
		starter.Add(1)
		var workers []worker.UserDeactivationNotificationWorker
		freq := time.Millisecond * 50
		for i := 0; i < 100; i++ {
			w := s.newUserDeactivationNotificationWorker(fmt.Sprintf("pod-%d", i), app)
			workers = append(workers, w)

			go func(w worker.UserDeactivationNotificationWorker) {
				starter.Wait()
				w.Start(freq)
			}(w)
		}
		starter.Done()
		// wait a few cycles before checking the results
		time.Sleep(freq * 5)
		for _, w := range workers {
			w.Stop()
		}
		time.Sleep(freq * 10) // give workers some time to stop for good
```

Reply:
@MatousJobanek done in 84bd48f

Reply:
and thanks for the suggestion 🙌